File: location-server/src/main/java/nl/lucien/domain/PathRepository.java
package nl.lucien.domain;

import nl.lucien.adapter.LocationDto;
import nl.lucien.adapter.PathQuery;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface PathRepository {

    Flux<Path> findAll();
    Mono<Path> findById(String pathId);
    Flux<Path> queryPaths(PathQuery pathQuery);
    Mono<Path> createPathFor(String userId);
    Mono<Path> addLocationToPath(String pathId, LocationDto locationDto);
    Mono<Path> deleteByPathId(String pathId);
}

File: location-cloud/src/main/kotlin/nl/cloud/location/domain/path/PathRepository.kt
package nl.cloud.location.domain.path

import nl.cloud.location.domain.user.UserId

interface PathRepository {
    fun fetchPathByUserId(userId: UserId): Path?
    fun createPath(userId: UserId): Path
    fun addLocationToPath(userId: UserId, location: TimedGeoLocation)
}